plugins/module_utils/openshift_adm_prune_auth.py

```diff
@@ -0,0 +1,339 @@
+        delete_options = client.V1DeleteOptions()
+        if propagation_policy:
+            delete_options.propagation_policy = propagation_policy
+
+        for namespace, name in candidates:
+            try:
+                result = resource.delete(name=name, namespace=namespace, body=delete_options)
+            except DynamicApiError as exc:
+                msg = "Failed to delete {kind} {namespace}/{name} due to: {msg}".format(kind=kind, namespace=namespace, name=name, msg=exc.body)
+                self.fail_json(msg=msg)
+            except Exception as e:
+                msg = "Failed to delete {kind} {namespace}/{name} due to: {msg}".format(kind=kind, namespace=namespace, name=name, msg=to_native(e))
+                self.fail_json(msg=msg)
+        return [y if x is None else x + "/" + y for x, y in candidates]
+
+    def update_resource_binding(self, ref_kind, ref_names, namespaced=False):
+
+        kind = 'ClusterRoleBinding'
+        api_version = "rbac.authorization.k8s.io/v1",
+        if namespaced:
+            kind = "RoleBinding"
+        resource = self.find_resource(kind=kind, api_version=api_version, fail=True)
+        result = resource.get(name=None, namespace=None).to_dict()
+        result = result.get('items') if 'items' in result else [result]
+
+        if len(result) == 0:
+            return [], False
+
+        def _update_user_group(binding_namespace, subjects):
+            users, groups = [], []
+            for x in subjects:
+                if x['kind'] == 'User':
+                    users.append(x['name'])
+                elif x['kind'] == 'Group':
+                    groups.append(x['name'])
+                elif x['kind'] == 'ServiceAccount':
+                    namespace = binding_namespace
+                    if x.get('namespace') is not None:
+                        namespace = x.get('namespace')
+                    if namespace is not None:
+                        users.append("system:serviceaccount:%s:%s" % (namespace, x['name']))
+            return users, groups
+
+        candidates = []
+        changed = False
+        for item in result:
+            subjects = item.get('subjects', [])
+            retainedSubjects = [x for x in subjects if x['kind'] == ref_kind and x['name'] in ref_names]
+            if len(subjects) != len(retainedSubjects):
+                updated_binding = item
+                updated_binding['subjects'] = retainedSubjects
+                binding_namespace = item['metadata'].get('namespace', None)
+                updated_binding['userNames'], updated_binding['groupNames'] = _update_user_group(binding_namespace, retainedSubjects)
+                candidates.append(binding_namespace + "/" + item['metadata']['name'] if binding_namespace else item['metadata']['name'])
+                if not self.check_mode:
+                    try:
+                        resource.apply(updated_binding, namespace=binding_namespace)
+                        changed = True
```

Comment:
This should be moved down to right after this `if` block. In check mode it would have changed.